lib/ansible/modules/windows/win_acl.ps1

```diff
@@ -141,7 +168,9 @@ If (Test-Path -Path $path -PathType Leaf) {
 ElseIf ($inherit -eq "") {
     $inherit = "ContainerInherit, ObjectInherit"
 }
- 
+
+$ErrorActionPreference = "Stop" 
```

Comment:
If entire blocks are being wrapped in a `try`, I'd suggest a hybrid approach; a pattern like this where we toggle it on/off as we enter the sections of code where we want to catch the exceptions.
```
If ($state -eq "present" -And $match -eq $false) {
        $ErrorActionPrference = "Stop"
        Try {
            $objACL.AddAccessRule($objACE)
            Set-ACL $path $objACL
            Set-Attr $result "changed" $true;
        }
        Catch {
            Fail-Json $result "an exception occurred when adding the specified rule - $_.Exception.Message"
        }
       finally {
           $ErrorActionPrference = "Continue"
       }
```
This is a small section of code for the example, but probably would be best to apply it at the outermost try/catch block if they are nested.

The rationale there is that if we care enough to wrap multiple statements in a try/catch block, we care enough to ensure that anything inside the block that fails should throw the exceptions.  

Limiting it to just the cmdlets opens up the possibility of letting non-terminating errors through the catch and having the module continue on and fail somewhere else and not be as helpful if the exception was caught at the point it fired.  PowerShell can skip the catch for non-terminating errors which can be undesirable.